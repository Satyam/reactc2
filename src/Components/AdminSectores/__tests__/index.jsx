import React from 'react';
import enzyme, { shallow, mount } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';

import AdminSectoresComponent, { Sector } from '../';

enzyme.configure({ adapter: new EnzymeAdapter() });

describe('Sector', () => {
  const sampleSector = {
    idSector: 'sampleSector',
    descr: 'descripción',
    descrCorta: 'descripción corta',
  };
  describe('snapshot testing', () => {
    // it('with no properties', () => {
    //   const wrapper = shallow(<Sector />);
    //   expect(toJson(wrapper)).toMatchSnapshot();
    // });
    it('with sector, checked', () => {
      const wrapper = shallow(<Sector sector={sampleSector} checked onChange={() => {}} />);
      expect(wrapper).toMatchSnapshot();
    });
    it('with sector, not checked', () => {
      const wrapper = shallow(<Sector sector={sampleSector} onChange={() => {}} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('event handling checking', () => {
    it('when unchecked and clicked, it should turn true', () => {
      const onChange = jest.fn();
      const preventDefault = jest.fn();
      const wrapper = mount(
        <Sector
          sector={{
            idSector: 'sampleSector',
            descr: 'descripción',
            descrCorta: 'descripción corta',
          }}
          onChange={onChange}
        />
      );
      wrapper.find('input').simulate('click', { preventDefault });
      expect(onChange.mock.calls).toEqual([['sampleSector', true]]);
      expect(preventDefault.mock.calls).toEqual([]);
    });
    it('when checked and clicked, it should turn false', () => {
      const onChange = jest.fn();
      const preventDefault = jest.fn();
      const wrapper = mount(
        <Sector
          sector={{
            idSector: 'sampleSector',
            descr: 'descripción',
            descrCorta: 'descripción corta',
          }}
          checked
          onChange={onChange}
        />
      );
      wrapper.find('input').simulate('click', { preventDefault });
      expect(onChange.mock.calls).toEqual([['sampleSector', false]]);
      expect(preventDefault.mock.calls).toEqual([]);
    });
  });
});

describe('AdminSectoresComponent', () => {
  const listSectores = [
    {
      idSector: 'constitucion',
      descrCorta: 'Constitución',
      descr: 'Estación Constitución, Ciudad de Buenos Aires, Argentina',
    },
    {
      idSector: 'retiro',
      descrCorta: 'Retiro',
      descr: 'Mentira, apenas una línea',
    },
  ];
  describe('snapshot testing', () => {
    it('with no properties', () => {
      const wrapper = shallow(<AdminSectoresComponent />);
      expect(wrapper).toMatchSnapshot();
    });
    it('with two sectors', () => {
      const wrapper = shallow(<AdminSectoresComponent sectores={listSectores} status={[]} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
