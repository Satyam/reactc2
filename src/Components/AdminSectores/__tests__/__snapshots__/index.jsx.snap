// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminSectoresComponent snapshot testing with no properties 1`] = `""`;

exports[`AdminSectoresComponent snapshot testing with two sectors 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Admin Sectores
    </title>
  </HelmetWrapper>
  <div
    className="form"
  >
    <ThemedList
      className="list"
      composeTheme="deeply"
      mapThemrProps={[Function]}
      ripple={false}
      selectable={false}
    >
      <Sector
        checked={false}
        key="constitucion"
        onChange={[Function]}
        sector={
          Object {
            "descr": "Estación Constitución, Ciudad de Buenos Aires, Argentina",
            "descrCorta": "Constitución",
            "idSector": "constitucion",
          }
        }
      />
      <Sector
        checked={false}
        key="retiro"
        onChange={[Function]}
        sector={
          Object {
            "descr": "Mentira, apenas una línea",
            "descrCorta": "Retiro",
            "idSector": "retiro",
          }
        }
      />
    </ThemedList>
    <ThemedNavigation
      actions={Array []}
      className="buttons"
      composeTheme="deeply"
      mapThemrProps={[Function]}
      routes={Array []}
      type="horizontal"
    >
      <ThemedThemed
        accept=".json"
        composeTheme="deeply"
        disabled={false}
        icon="file_upload"
        label="Agregar"
        mapThemrProps={[Function]}
        name="archivo"
        onChange={[Function]}
        raised={true}
        ripple={true}
        rippleCentered={false}
        rippleClassName=""
        rippleMultiple={true}
        rippleSpread={2}
      />
      <ThemedThemed
        composeTheme="deeply"
        disabled={true}
        icon="delete"
        label="Delete"
        mapThemrProps={[Function]}
        name="delete"
        onClick={[Function]}
        raised={true}
        ripple={true}
        rippleCentered={false}
        rippleClassName=""
        rippleMultiple={true}
        rippleSpread={2}
      />
    </ThemedNavigation>
  </div>
</div>
`;

exports[`Sector snapshot testing with sector, checked 1`] = `
<ThemedListCheckbox
  caption="descripción corta"
  checked={true}
  composeTheme="deeply"
  disabled={false}
  legend="descripción"
  mapThemrProps={[Function]}
  onChange={[Function]}
/>
`;

exports[`Sector snapshot testing with sector, not checked 1`] = `
<ThemedListCheckbox
  caption="descripción corta"
  checked={false}
  composeTheme="deeply"
  disabled={false}
  legend="descripción"
  mapThemrProps={[Function]}
  onChange={[Function]}
/>
`;
